File: mobile/lib/models/report.dart
import '../utils/formatters.dart';

class Report {
  final int id;
  final int? userId;
  final String? periodStart;
  final String? periodEnd;
  final String content;
  final int calls;
  final int visits;
  final int emails;
  final String status;
  final String? reviewComment;
  final String? userName;
  final String? reviewedByName;
  final String? createdAt;

  const Report({
    required this.id,
    this.userId,
    this.periodStart,
    this.periodEnd,
    required this.content,
    this.calls = 0,
    this.visits = 0,
    this.emails = 0,
    this.status = 'en_attente',
    this.reviewComment,
    this.userName,
    this.reviewedByName,
    this.createdAt,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: toInt(json['id']) ?? 0,
      userId: toInt(json['user_id']),
      periodStart: json['period_start'] as String?,
      periodEnd: json['period_end'] as String?,
      content: json['content'] as String? ?? '',
      calls: toInt(json['calls']) ?? 0,
      visits: toInt(json['visits']) ?? 0,
      emails: toInt(json['emails']) ?? 0,
      status: json['status'] as String? ?? 'en_attente',
      reviewComment: json['review_comment'] as String?,
      userName: json['user_name'] as String?,
      reviewedByName: json['reviewed_by_name'] as String?,
      createdAt: json['created_at'] as String?,
    );
  }
}

File: client/src/models/report.js
/**
 * Modèle de l'entité Rapport (rapport d'activité).
 */
export const REPORT_FIELDS = {
  id: 'number',
  user_id: 'number',
  period_start: 'string (date ISO)',
  period_end: 'string (date ISO)',
  content: 'string',
  calls: 'number',
  visits: 'number',
  emails: 'number',
  status: 'string (brouillon | soumis | valide)',
  reviewed_by: 'number|null',
  review_comment: 'string|null',
  created_at: 'string (date ISO)',
  updated_at: 'string (date ISO)',
};

/**
 * Fabrique un rapport avec des valeurs par défaut.
 * @param {object} data - valeurs initiales (optionnel)
 * @returns {object} une structure rapport
 */
export function createReport(data = {}) {
  return {
    id: null,
    user_id: null,
    period_start: null,
    period_end: null,
    content: '',
    calls: 0,
    visits: 0,
    emails: 0,
    status: 'brouillon',
    reviewed_by: null,
    review_comment: null,
    created_at: null,
    updated_at: null,
    ...data,
  };
}

export default createReport;
